android/java/org/chromium/chrome/browser/crypto_wallet/activities/NftDetailActivity.java

```diff
@@ -133,31 +152,59 @@ protected void triggerLayoutInflation() {
                             ? defaultNetwork.blockExplorerUrls[0]
                             : "";
 
-                    String tokenStr = String.format(TOKEN_ID_FORMAT, mNftTokenId);
-                    SpannableString spannable = new SpannableString(tokenStr);
-
-                    if (!TextUtils.isEmpty(blockExplorerUrl)) {
-                        String url = String.format(
-                                NFT_URL_FORMAT, blockExplorerUrl, mContractAddress, mNftTokenId);
-                        NoUnderlineClickableSpan linkSpan =
-                                new NoUnderlineClickableSpan(this, R.color.brave_link,
-                                        (textView) -> { TabUtils.openLinkWithFocus(this, url); });
-                        spannable.setSpan(
-                                linkSpan, 0, spannable.length(), Spanned.SPAN_EXCLUSIVE_EXCLUSIVE);
+                    SpannableString spannable;
+                    if (mIsErc721) {
+                        String tokenStr = String.format(Locale.US, TOKEN_ID_FORMAT, mNftTokenId);
+                        spannable = new SpannableString(tokenStr);
+
+                        if (!TextUtils.isEmpty(blockExplorerUrl)) {
+                            String url = String.format(Locale.US, NFT_ERC721_URL_FORMAT,
+                                    blockExplorerUrl, mContractAddress, mNftTokenId);
+                            createClickableLink(blockExplorerUrl, url, spannable);
+                        }
+
+                    } else {
+                        String mintAddress = Utils.getTruncatedAddress(mContractAddress);
+                        spannable = new SpannableString(mintAddress);
+
+                        if (!TextUtils.isEmpty(blockExplorerUrl)) {
+                            // Blockchain explorer URLs may contain a cluster endpoint.
+                            // When present it must be appended at the end of the formatted URL.
+                            String[] splitBlockExplorerUrl = blockExplorerUrl.split("/\\?cluster=");
+                            String baseUrl = splitBlockExplorerUrl[0];
+                            String url;
+                            if (splitBlockExplorerUrl.length > 1) {
+                                String cluster = splitBlockExplorerUrl[1];
+                                url = String.format(Locale.US, NFT_SPL_URL_FORMAT_WITH_CLUSTER,
+                                        baseUrl, mContractAddress, cluster);
+                            } else {
+                                url = String.format(
+                                        Locale.US, NFT_SPL_URL_FORMAT, baseUrl, mContractAddress);
```

Comment:
nit: Isn't better to use [Locale.ENGLISH](https://docs.oracle.com/javase/8/docs/api/java/util/Locale.html#ENGLISH) for generic use?

Reply:
It will not change much in this particular case. `Locale.ENGLISH` would have been my first choice. Right now there's a good mix in the codebase between `Locale.ENGLISH` and `Locale.US`. It seems though `Locale.US` was the locale adopted in the latest changes. I'll change it to `ENGLISH`.